flags/src/main/java/com/yahoo/vespa/flags/json/RelationalCondition.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.flags.json;
+
+import com.yahoo.component.Version;
+import com.yahoo.vespa.flags.FetchVector;
+import com.yahoo.vespa.flags.json.wire.WireCondition;
+
+import java.util.function.Predicate;
+
+/**
+ * @author hakonhall
+ */
+public class RelationalCondition implements Condition {
+    private final RelationalPredicate relationalPredicate;
+    private final Predicate<String> predicate;
+    private final FetchVector.Dimension dimension;
+
+    public static RelationalCondition create(CreateParams params) {
```

Comment:
Make this into constructor like the others/change the others to use factory method?

Reply:
Made factories.